includes/Core/Notifications/Notification.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Notifications\Notification
+ *
+ * @package   Google\Site_Kit\Core\Notifications
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Notifications;
+
+/**
+ * Class for representing a notification.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Notification {
+	/**
+	 * Unique notification slug.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Notification arguments.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $args;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug Unique notification slug.
+	 * @param array  $args {
+	 *     Associative array of notification arguments.
+	 *
+	 *     @type string $title            Required notification title.
+	 *     @type string $content          Required notification content. May contain inline HTML tags.
+	 *     @type string $image            Image URL.
+	 *     @type string $cta_url          Call to action URL.
+	 *     @type string $cta_label        Call to action anchor text.
+	 *     @type string $cta_target       Call to action anchor target.
+	 *     @type string $learn_more_url   Learn more URL.
+	 *     @type string $learn_more_label Learn more anchor text.
+	 *     @type bool   $dismissible      Whether the notice should be dismissible. Default false.
+	 *     @type string $dismiss_label    Dismiss anchor text.
+	 * }
+	 */
+	public function __construct( $slug, array $args ) {
+		$this->slug = (string) $slug;
+		$this->args = array_merge(
+			array(
+				'title'            => '',
+				'content'          => '',
```

Comment:
Since these are always empty-string by default, any reason the component does `notification.content || ''`  when passing props? Seems unneeded. 🤷‍♂ Belt and braces I guess?

Reply:
This ensures that the keys are always set on the PHP side.